lib/semantics/pointer-assignment.cc

```diff
@@ -180,7 +184,9 @@ void PointerAssignmentChecker::Check(const evaluate::Designator<T> &d) {
     context_.messages().Say("Pointer target is not a named entity"_err_en_US);
     return;
   }
-  std::optional<parser::MessageFixedText> msg;
+  std::variant<std::monostate, parser::MessageFixedText,
```

Comment:
Might be more clear if `std::optional<std::variant<parser::MessageFixedText, parser::MessageFormattedText>>`, avoiding the use of `std::monostate`.

Reply:
OK